sdk/storage_datalake/src/operations/path_head.rs

```diff
@@ -87,18 +87,17 @@ pub struct HeadPathResponse {
     pub common_storage_response_headers: CommonStorageResponseHeaders,
     pub etag: String,
     pub last_modified: DateTime<Utc>,
-    pub properties: Option<Properties>,
```

Comment:
Hi Cameron, thanks for the response!

This needs to be an `Option` because `Properties`, in this context, is equal to `x-ms-properties`, and this header __only__ gets returned when you don't specify an `action`.  See the docs [here](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/get-properties). The previous implementation didn't consider it _optional_ because _GetProperties_ was the only implementation that existed.  However, to support GetStatus (a derivation of GetProperties), it will need to be optional to support both. Hence the reason for changing this to an `Option`.  Without this, the API response cannot be processed correctly.  Take a look at the test I wrote for this function for context, specifically [here ](https://github.com/rickrain/azure-sdk-for-rust/blob/c2d32b7a49715c54b52dcef0854ce7ca1c280843/sdk/storage_datalake/tests/files.rs#L211)and [here](https://github.com/rickrain/azure-sdk-for-rust/blob/c2d32b7a49715c54b52dcef0854ce7ca1c280843/sdk/storage_datalake/tests/files.rs#L215).  You will see that when the `action` is `GetStatus`, that `x-ms-properties` is not returned.